platform/sun50i/devices.c

```diff
@@ -54,3 +67,13 @@ static struct device r_intc = {
 	.drv     = &sun4i_intc_driver,
 	.drvdata = SUN4I_INTC_DRVDATA { { 0 } },
 };
+
+static struct device r_pio = {
+	.name     = "r_pio",
+	.clock    = CCU_GATE(R_CCU_GATE_R_PIO),
+	.clockdev = &r_ccu,
+	.drv      = &sunxi_pio_driver,
+	.drvdata  = BIT(0), /*< Physically implemented R_PIO ports (0). */
+	.irq      = IRQ_R_PIO_PL,
+	.regs     = DEV_R_PIO,
```

Comment:
Same issue with `.regs` ordering as above.

Also, to have the `.irq`, you need an `.irqdev = &r_pio,`. Or alternatively, since we're not using the IRQs yet, you can omit both.

Reply:
Fixed.